src/internal/testutil/kubeclient.go

```diff
@@ -190,7 +224,7 @@ func WaitForPachdReady(t testing.TB, namespace string) {
 		time.Sleep(time.Second * 5)
 	}
 	watch, err := k.CoreV1().Pods(namespace).Watch(context.Background(), metav1.ListOptions{
-		LabelSelector: "app=pachd",
+		LabelSelector: fmt.Sprintf("app=%s", appName),
```

Comment:
Huh, I really thought it was used, but I must have gotten confused there. That changes a lot. 

Would these changes work for you?
- In kubeclient.go delete everything except `GetKubeClient`, and `DeletePipelineRC`(I'll make a ticket and assign to myself to revisit `DeletePipelineRC`, I think it has a similar issue to DeletePods)
- Move the delete pod and readiness wait logic into the tests
- For logic on readiness, I was planning to follow the logic of minikubeTestEnv.waitForPachd, but I think it does still use the readiness check instead of sending a request to a new port: `p.Status.Phase == v1.PodRunning && strings.HasSuffix(p.Spec.Containers[0].Image, ":"+version) && p.Status.ContainerStatuses[0].Ready && len(pachds.Items) == 1`